OpenRA.Game/Orders/UnitOrderGenerator.cs

```diff
@@ -79,11 +79,19 @@ public string GetCursor(World world, CPos xy, MouseInput mi)
 				target = frozen != null ? Target.FromFrozenActor(frozen) : Target.FromCell(world, xy);
 			}
 
-			var orders = world.Selection.Actors
+			string cursorName = null;
+
+			var ordersWithCursor = world.Selection.Actors
 				.Select(a => OrderForUnit(a, target, mi))
-				.Where(o => o != null);
+				.Where(o => o != null && o.Cursor != null);
+
+			if (ordersWithCursor.Any())
```

Comment:
The main two groups of extension methods are in  `Exts.cs` and `StreamExts.cs` in the OpenRA.Game project.